fission-server/src/models/account.rs

```diff
@@ -154,6 +155,13 @@ impl Account {
         conn: &mut Conn<'_>,
         cid: String,
     ) -> Result<NewVolumeRecord, diesel::result::Error> {
+        let ipfs = ipfs_api::IpfsClient::default();
+
+        if ipfs.pin_add(&cid, true).await.is_err() {
+            // FIXME: Use better error
+            return Err(diesel::result::Error::NotFound);
```

Comment:
This error is confusing can we do something better?

Reply:
It looks like the error that comes back is described by [this](https://docs.rs/ipfs-api/latest/ipfs_api/enum.Error.html) enum, so if you wanted, you could define a `From` instance from that type to our error type, and differentiate between server and client errors.

Reply:
Yeah makes sense. This method is using the diesel error type though, I assume we want something different here then? `anyhow::Error`?

Reply:
You'd potentially want to define your own error type and then define a translation from that to `AppError`, with that conversion happening in the route! You could also just return an `AppError` directly for now, though that does muddle the boundary between web and non-web concerns slightly.